solar_system.rb

```diff
@@ -0,0 +1,385 @@
+# This program starts its first command on line 370....
+
+#########################################################################################################
+# HARD-TYPED DATA:
+planets_array =
+[{ name: "Mercury",
+  year_length: 88, #Earth days
+  distance_from_the_sun: 35.98,  #million miles
+  planet_type: "Terrestrial",
+  number_of_moons: 0,
+  interesting_fact1: "Mercury has a solid, cratered surface, much like the Earth's moon.",
+  interesting_fact2: "Mercury is only slightly larger than Earth's moon" #
+},
+{ name: "Venus",
+  year_length: 225,
+  distance_from_the_sun: 67.24 ,
+  planet_type: "Terrestrial",
+  number_of_moons: 0,
+  interesting_fact1: "hottest planet in our solar system",
+  interesting_fact2: "Venus has mountains, valleys, and tens of thousands of volcanoes"
+},
+{ name: "Earth",
+  year_length: 365,
+  distance_from_the_sun: 93 ,
+  planet_type: "Terrestrial",
+  number_of_moons: 1,
+  interesting_fact1: "Earth is the biggest of the terrestrial planets",
+  interesting_fact2: "The name Earth is at least 1,000 years old"
+},
+{ name: "Mars",
+  year_length: 687,
+  distance_from_the_sun: 142 ,
+  planet_type: "Terrestrial",
+  number_of_moons: 2,
+  interesting_fact1: "Mars has two moons named Phobos and Deimos.",
+  interesting_fact2: "Mars is a dynamic world that additionally has seasons, polar ice caps, volcanoes, canyons and weather"
+},
+{ name: "Jupiter",
+  year_length: 4333,
+  distance_from_the_sun: 484,
+  planet_type: "Gas Giant",
+  number_of_moons: 53,
+  interesting_fact1: "Some of Jupiter's moons have oceans beneath their crusts that might support life",
+  interesting_fact2: "Jupiter has 53 known moons, with many additional moons to be confirmed and named"
+},
+{ name: "Saturn",
+  year_length: 10759,
+  distance_from_the_sun: 886,
+  planet_type: "Gas Giant",
+  number_of_moons: 60,
+  interesting_fact1: "Like fellow gas giant Jupiter, Saturn is a massive ball made mostly of hydrogen and helium",
+  interesting_fact2: "Adorned with thousands of beautiful ringlets, Saturn is unique among the planets"
+  }]
+
+  #########################################################################################################
+  # CLASS FOR SOLAR SYSTEM
+
+  class SolarSystem
+
+    attr_accessor  :solar_system
+
+    def initialize(planets_array)
+      @solar_system = planets_array
+      @names = []
```

Comment:
Does this need to be an instance variable?  It looks like it could be used instead as a local variable in `create_names`.